File: Scala/src/test/sorts/MergeSortTest.scala
package sorts

import com.ckeyser.sorts.MergeSort
import com.ckeyser.util.SortUtil
import org.scalatest.FunSpec
import util.TestUtil

import scala.reflect.ClassTag

/**
  * Created by Chris on 10/2/2016.
  */
class MergeSortTest extends FunSpec{
    it("should correctly sort an array") {
        val arr = TestUtil.RandomArray(TestUtil.ARR_SIZE)

        val result = MergeSort.mergesort(arr)

        assert(SortUtil.isSorted(result))
    }
}

File: Java/test/sorts/MergeSortTest.java
package sorts;

import org.junit.Assert;
import org.junit.Test;
import util.SortUtil;
import util.TestUtil;

/**
 * Created by Chris Keyser on 10/3/2016.
 */
public class MergeSortTest {
    @Test
    public void MergeSort_SortsCorrectly() {
        int arr[] = TestUtil.generateRandomArray(TestUtil.ARR_SIZE);

        int[] sorted = MergeSort.sort(arr);

        Assert.assertTrue(SortUtil.isSorted(sorted));
    }
}
